### PR TITLE
Pushing Dockerfile from feature to master

### PR DESCRIPTION
**Pull request by ykhandev**
Pushing Dockerfile from feature to master

### CODE REVIEW

**Comment by dan-hill2802** [comment]
Can you remove evrything from the readme except steps 7 and 8. You can now update step 7 with the image name

**Comment by ykhandev** [comment]
Updated the README @dan-hill2802

File: README.md [hunk 1]
```diff
@@ -1,2 +1,11 @@
 # docker-spark-gpg
-Docker Alpine image with Spark, Python, Java and gpg. Used for local dev environment.
+Docker Alpine image with Spark, Python, Java and gpg. Used for loacal dev environment.
+
+Instructions
+
+7. Run the command in order to mount the export data folder on to the root of the Docker container
```
Comment (dan-hill2802) [inline README.md]: This should mention the 3 directories that need mounting, also renumber these, 7 and 8 no longer make sense

File: README.md [hunk 2]
```diff
@@ -1,2 +1,11 @@
...
+`docker run -v ~/export-data:/root/export-data -v ~/export-data/gpg:/root/gpg -it docker-spark-gpg :latest bash`
```
Comment (dan-hill2802) [inline README.md]: I know it's only an example, but got should not be a sub dir of export-data 
Shouldn't there be a mount point for the output?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,11 @@
...
+
+8. You should now be inside the docker container and be able to see the export-data directory
+
```
Comment (dan-hill2802) [inline README.md]: Should also mention here that the gpg keys in the mounted directory will have already been imported

Comment (ykhandev) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -1,2 +1,34 @@
...
+Instructions
+
+
+1. Run the command in order to mount the export-data folder , the TestData folder and the gpg folder on to the root of the Docker container
+
+'docker run -v /Users/yasserkhan/Desktop/docker-spark-gpg/export-data:/root/export-data -v /Users/yasserkhan/Desktop/docker-spark-gpg/Spark:/root/Spark -v /Users/yasserkhan/Desktop/docker-spark-gpg/gpg:/root/gpg -it docker-spark-gpg :latest bash'
+
+
+NOTE: The gpg keys in the mounted directory will have already been imported. Use the command 'gpg --list-keys' to see a list of all keys imported. There will be a total of 4
```
Comment (dan-hill2802) [inline README.md]: can you remove `There will be a total of 4` as that entirely depends on how many are in the mounted dir

File: README.md [hunk 5]
```diff
@@ -1,2 +1,34 @@
 # docker-spark-gpg
-Docker Alpine image with Spark, Python, Java and gpg. Used for local dev environment.
+Docker Alpine image with Spark, Python, Java and gpg. Used for loacal dev environment.
+
+Instructions
+
+
+1. Run the command in order to mount the export-data folder , the TestData folder and the gpg folder on to the root of the Docker container
+
+'docker run -v /Users/yasserkhan/Desktop/docker-spark-gpg/export-data:/root/export-data -v /Users/yasserkhan/Desktop/docker-spark-gpg/Spark:/root/Spark -v /Users/yasserkhan/Desktop/docker-spark-gpg/gpg:/root/gpg -it docker-spark-gpg :latest bash'
```
Comment (dan-hill2802) [inline README.md]: this is too tightly coupled to you local directory structure. The following would be better:
'docker run -v ~/export-data:/root/export-data -v ~/Spark:/root/Spark -v ~/gpg:/root/gpg -it danhill2802/spark-gpg :latest bash'
Also note, the container name isn't `docker-spark-gpg`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,34 @@
...
+
+
+NOTE: The gpg keys in the mounted directory will have already been imported. Use the command 'gpg --list-keys' to see a list of all keys imported. There will be a total of 4
+
+
+2. You should now be inside the docker container and be able to see the export-data directory
+
+3. To call this for any single collection we would use:
```
Comment (dan-hill2802) [inline README.md]: Can you remove this line and everything that follows it? This is a specific use case, so is more suited to the Confluence page than the README

File: README.md [hunk 7]
```diff
@@ -1,2 +1,34 @@
 # docker-spark-gpg
-Docker Alpine image with Spark, Python, Java and gpg. Used for local dev environment.
+Docker Alpine image with Spark, Python, Java and gpg. Used for loacal dev environment.
+
```
Comment (dan-hill2802) [inline README.md]: Can you add a line to say that the docker container is automatically published to https://hub.docker.com/r/danhill2802/spark-gpg/

Comment (ykhandev) [reply README.md]: Updated README

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,44 @@
+### 1. Get Linux
+FROM alpine:3.7
+
+ADD gpg /root/gpg
```
Comment (dan-hill2802) [inline Dockerfile]: This cause the build to fail if the `gpg` dir is missing. Also, this isn't needed as you have mounted in your run example

File: README.md [hunk 9]
```diff
@@ -1,2 +1,17 @@
...
+The docker container is automatically published to https://hub.docker.com/r/danhill2802/spark-gpg/
+
+Instructions
+
+
+1. Run the command in order to mount the export-data folder , the TestData folder and the gpg folder on to the root of the Docker container
+
+'docker run -v ~/export-data:/root/export-data -v ~/Spark:/root/Spark -v ~/gpg:/root/gpg -it danhill2802/spark-gpg :latest bash'
```
Comment (dan-hill2802) [inline README.md]: remove the space in `danhill2802/spark-gpg :latest`

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+	if [ ! -e /usr/bin/pip ]; then ln -s pip2 /usr/bin/pip ; fi && \
+	if [[ ! -e /usr/bin/python ]]; then ln -sf /usr/bin/python2.7 /usr/bin/python; fi && \
+	rm -r /root/.cache \
+### 4. Get pypandoc ,pyyaml and pyspark from pip
+	&& pip  install --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org pyyaml \
+	&& pip  install --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org pypandoc \
+	&& pip install --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org  pyspark 
+
+### 4. Run GPG command within the gpg folder in order to import all the keys in to the container
+RUN gpg --import *.asc
```
Comment (dan-hill2802) [inline Dockerfile]: This needs to be a `CMD` as it needs to be run when the container starts, not when the image is built

**Pull request merged**